lib/config/fileconf.go

```diff
@@ -198,7 +200,14 @@ func MakeSampleFileConfig(flags SampleFlags) (fc *FileConfig, err error) {
 		g.DataDir = defaults.DataDir
 	}
 
-	g.AuthToken = flags.AuthToken
+	if flags.JoinMethod == "" {
+		flags.JoinMethod = "token"
+	}
```

Comment:
You could add a `.Default("token")` to the flag declaration and remove this block.